### DISCUSSION TITLE
Contributors count on the website doesn't match with the API

### DISCUSSION THREAD

**Discussion by bilgeyucel** [community/community#63501] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I use `https://api.github.com/repos/:org/:repo/contributors?per_page=100&page=1` endpoint to get the number of contributors of the repo. On Github repo UI, it shows 195 contributors, whereas I can only get 186 through the endpoint. It would be nice if these numbers could match. Is there a solution to this?

**Comment by Akashh2151** [comment]
Discrepancies between the number of contributors shown on the GitHub repository UI and the number returned by the GitHub API's contributors endpoint can occur due to a few reasons:

Unmerged Contributions: The GitHub API's contributors endpoint only includes contributors whose pull requests have been merged into the repository's default branch (usually main or master). Contributors with unmerged pull requests, even if they've made significant contributions, may not be included in the API response.

Anonymous Contributors: If contributors have made contributions without associating them with a GitHub account (i.e., anonymous contributions), they might not appear in the GitHub API response. However, the GitHub UI might still count these contributions for the total contributor count.

Rate Limiting and Pagination: When fetching contributors from the GitHub API, make sure you're correctly handling pagination using the page parameter. GitHub API responses are paginated, and if you're only fetching the first page of contributors, you might not get an accurate count. Use the Link header in the API response to determine if there are additional pages of contributors.

Caching: The GitHub UI might use different caching mechanisms or update intervals compared to direct API requests. This can lead to differences in the displayed contributor count.

Data Synchronization: The data displayed on the GitHub UI might take some time to synchronize with the underlying data, including contributor statistics. This can lead to temporary differences between the UI and API results.

If you're looking to get the most accurate and up-to-date contributor count, consider implementing a mechanism that uses both the GitHub API and the GitHub UI. You can use the API to fetch the contributors and their contributions, and you can cross-reference this data with the contributor list shown on the GitHub UI.

Also, remember that the specific details and behaviors of the GitHub API can change over time, so it's a good idea to periodically review the GitHub API documentation and any updates they make to ensure you're using the most accurate methods for retrieving contributor data.
